### ISSUE TITLE
Adding support for JsonPointers

### ISSUE THREAD

**Issue by RiccardoRossi** [KratosMultiphysics/Kratos#11428]
**Description**
let's imagine that we have the following

~~~json
{
   "problem_data" : {
        "start_time" : 0.0,
        "end_time" : 1.0
    }
    "other_settings" : {
         "end_time" : 1.0
    }  
}
~~~

where for any reason "end_time" is needed in two blocks of the json. In essence i would propose to replace that code by something like

~~~json
{
   "problem_data" : {
        "start_time" : 0.0,
        "end_time" : 1.0
    }
    "other_settings" : {
         "end_time" : { "@json_pointer" : "/problem_data/end_time" } 
    }  
}
~~~

so that one only needs to input the "end_time" once and the value is propagated to the rest of the code.

In here i used the standard definition of "json pointer" as described here

"[LINK](https://json.nlohmann.me/features/json_pointer/)"

In the design i propose this json pointers would be resolved ONLY when the json is parsed from a string, moment at which we would replace it by the actual value.

as a matter of fact, i would also add to the interface a function "GetPointer" so that one could write

~~~py
my_json.GetPointer("/problem_data/end_time")
~~~

instead of 

~~~py
my_json["problem_data"]["end_time"]
~~~

however this is obviosuly not too urgent

**Comment by matekelemen** [comment]
we can already do this with `@include_json` (available since #10004), though it expects another file, not a value in the current json.

**Comment by RiccardoRossi** [comment]
yes, i used the same syntax on purpose, but this is a pointer to an entry within the same json

**Comment by RiccardoRossi** [comment]
also note that in a sense here you replace a "value" instead of the key

**Comment by RiccardoRossi** [comment]
Hi @matekelemen first of all let me comment that i did not invent this. Json pointers are actually standardized as shortcuts to values (read it as "ways to access to locations within the json")

yet your point is valid.
my intention was to avoid any runtime cost other than in the initial parsing. 
To achieve this i  did not want the "pointer" to behave as a actual pointer (possibly we should then call it **@copied_value** or something of the sort) but only as a way for someone to refer ONCE to another location of the code. 

that is, after parsing, 

~~~json
{
   "problem_data" : {
        "start_time" : 0.0,
        "end_time" : 1.0
    }
    "other_settings" : {
         "end_time" : { "@json_pointer" : "/problem_data/end_time" } 
    }  
}~~~

would become in memory

~~~json
{
   "problem_data" : {
        "start_time" : 0.0,
        "end_time" : 1.0
    }
    "other_settings" : {
         "end_time" : 1.0
    }  
}

if after the initial parsing you change the "end_time" in the root, the change would NOT be reflected into the other "end_time".


~~~

**Comment by pooyan-dadvand** [comment]
@KratosMultiphysics/technical-committee agree with this addition as it can be useful in scenarios like several repeated values in the json file.

**Comment by loumalouomega** [comment]
It is already in the library: https://github.com/KratosMultiphysics/Kratos/blob/2941a013270e4b0ee0bf36259ec8e3048387b5d9/external_libraries/json/json.hpp#L13758

**Comment by ddiezrod** [comment]
I find this super useful in many of our cases! There is only one thing that I'd be missing: being able to point values from a different file (as @RiccardoRossi said, its not the same pointing to a value as including a json)
